### PR TITLE
Explicit resolution state saving

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

**Comment by kasper-piskorski** [comment]
please retest this please

**Comment by kasper-piskorski** [comment]
please retest this please - ai.grakn.engine.controller.UserControllerTest.ai.grakn.engine.controller.UserControllerTest

**Comment by kasper-piskorski** [comment]
!rtg - ai.grakn.engine.controller.UserControllerTest.ai.grakn.engine.controller.UserControllerTest

**Comment by kasper-piskorski** [comment]
please retest this please  ai.grakn.engine.controller.UserControllerTest.ai.grakn.engine.controller.UserControllerTest

File: grakn-test/src/test/java/ai/grakn/test/property/ThingPropertyTest.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 @RunWith(JUnitQuickcheck.class)
 public class ThingPropertyTest {
 
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/property/ThingPropertyTest.java]: I have to ask did you create a TP bug report to get this test fixed?

Comment (kasper-piskorski) [inline grakn-test/src/test/java/ai/grakn/test/property/ThingPropertyTest.java]: Ok will do.

Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/property/ThingPropertyTest.java]: Add me as the dev on that bug please.

Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/property/ThingPropertyTest.java]: Thanks Kasper: https://work.grakn.ai/entity/16317

**Pull request merged**